configstack/stack_plan.go

```diff
@@ -0,0 +1,135 @@
+package configstack
+
+import (
+	"fmt"
+	"github.com/gruntwork-io/terragrunt/options"
+	"github.com/gruntwork-io/terragrunt/shell"
+	"github.com/gruntwork-io/terragrunt/util"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+// The returned information for each module
+type moduleResult struct {
+	Module    TerraformModule
+	Err       error
+	Message   string
+	NbChanges int
+}
+
+func (stack *Stack) planWithSummary(terragruntOptions *options.TerragruntOptions) error {
+	// We do a special treatment for -detailed-exitcode since we do not want to interrupt the processing of dependant
+	// stacks if one dependency has changes
+	detailedExitCode := util.ListContainsElement(terragruntOptions.TerraformCliArgs, "-detailed-exitcode")
+	if detailedExitCode {
+		util.RemoveElementFromList(terragruntOptions.TerraformCliArgs, "-detailed-exitcode")
+	}
+
+	results := make([]moduleResult, 0, len(stack.Modules))
+	err := RunModulesWithHandler(stack.Modules, getResultHandler(detailedExitCode, &results), NormalOrder)
+	printSummary(terragruntOptions, results)
+
+	// If there is no error, but -detail-exitcode is specified, we return an error with the number of changes.
+	if err == nil && detailedExitCode {
+		sum := 0
+		for _, status := range results {
+			sum += status.NbChanges
+		}
+		err = countError{sum}
+	}
+
+	return err
+}
+
+// Returns the handler that will be executed after each completion of `terraform plan`
+func getResultHandler(detailedExitCode bool, results *[]moduleResult) ModuleHandler {
+	return func(module TerraformModule, output string, err error) error {
+		inspectPlanResult(module, output)
+		if exitCode, convErr := shell.GetExitCode(err); convErr == nil && detailedExitCode && exitCode == 2 {
+			err = nil
```

Comment:
I will add a comment.

The logic is simply that we do not want to consider exit code 2 as an error and not execute the dependants because there is an "error" in the dependencies. 2 is not an error in this case, it is simply a status. We reintroduce the exit code at the very end to mimic the behaviour of the `terrafrom plan -detailed-exitcode` to exit with 2 if there are changes in any of the module in the stack.